portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -101,21 +105,36 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
     long timestamp = System.currentTimeMillis();
+    float sentimentScore = getSentimentScore(newComment);
     
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("content", newComment);
     commentEntity.setProperty("timestamp", timestamp);
     commentEntity.setProperty("likeCount", 0);
     commentEntity.setProperty("email", userEmail);
+    commentEntity.setProperty("sentimentScore", sentimentScore);
     this.datastore.put(commentEntity);
   }
 
   @Override
   public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query q = new Query("Comment").setKeysOnly();
     PreparedQuery pq = this.datastore.prepare(q);
-    for (Entity entity : pq.asIterable()){
+    for (Entity entity : pq.asIterable()) {
       this.datastore.delete(entity.getKey());
     }
   }
+  
+  private float getSentimentScore(String msg) throws IOException {
+    // Use Sentiment Analysis API
+    Document doc =
+      Document.newBuilder().setContent(msg).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
+    // The score is a float number between -1 and 1.
+    // The greater the score is, the more positive the msg is.
```

Comment:
Useful information! It would be nicer if it was above the method.

Reply:
Done 👌